cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/SourceControlManagementService.java

```diff
@@ -229,8 +236,15 @@ public ApplicationRecord pullAndDeploy(ApplicationReference appRef) throws Excep
     AppRequest<?> appRequest = pullResponse.getAppRequest();
     SourceControlMeta sourceControlMeta = new SourceControlMeta(pullResponse.getApplicationFileHash());
     // Deploy with a generated uuid
-    String versionId = RunIds.generate().getId();
-    ApplicationWithPrograms app = appLifecycleService.deployApp(appRef.app(versionId), appRequest,
+    ApplicationId appId = appRef.app(RunIds.generate().getId());
```

Comment:
It seems that, per https://github.com/cdapio/cdap/blob/912ceb4204ece28a55bd6dc8dd490b81e14a77f1/cdap-proto/src/main/java/io/cdap/cdap/proto/id/ApplicationReference.java#L52, `appRef.app(String version)` takes a string which is the version. Can you explain why we're passing the run ID here?

Reply:
Sorry for the confusion. It was brought in LCM work that we use `RunIds.generate().getId()` to create the uuid version.